### ISSUE TITLE
piped commands via ssh?

### ISSUE THREAD

**Issue by git-os** [amoffat/sh#713]
```python
import sh
print(sh.awk("{print $5}", _in=sh.df("-h", "/")))
```
works fine but none of these seem to work

```python
import sh
srv = sh.ssh("root@mysrv")
print(srv.awk("{print $5}", _in=sh.df("-h", "/")))
print(srv.awk("{print $5}", _in=srv.df("-h", "/")))
```
is this intended behaviour because it is not supported?

**Comment by amoffat** [comment]
Yes intended behavior. `srv` is a string (the return result of `sh.ssh`). What you're trying to do, as simple as it appears conceptually, is very advanced. You'll need to use a different library for running remote processes.

**Comment by git-os** [comment]
Got it, maybe you could give me advice about implementation. Currently I have a class, that pretty much gets a binary along with its args (including potentially included _in special kwargs). Could I use contrib wrapping somehow to maybe execute the first level (remote) Command and then have a follow up function passing the  result of it to the _in defined command and execute it locally, then return the result?

**Comment by ecederstrand** [comment]
Keeping SSH connections open and running remote commands on a server is outside the scope of sh. It's not that it can't be done, but sh doesn't provide any help besides running commands locally and collecting the output. If you want to automate remote execution of commands, a tool like Fabric is a better choice.

**Comment by git-os** [comment]
ok, thanks
